letsencrypt/models/base_config_settings.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+import re
+from inspect import getmembers, isfunction
+from transip.service.domain import DomainService
+
+
+class BaseConfigSettings(models.TransientModel):
+    _inherit = 'base.config.settings'
+
+    dns_provider = fields.Selection(selection='_get_supported_dns_providers')
+    dns_function = fields.Char(
+        help='Holds the function name for the DNS updates')
+    dns_login = fields.Char()
+    dns_key = fields.Char()
+    dns_url = fields.Char()
+
+    def _get_supported_dns_providers(self):
+        providers = []
+        for func in [x for x in getmembers(self)
```

Comment:
read the [doc](https://docs.python.org/2/library/inspect.html#inspect.getmembers) of functions you use (meaning pass the below as lambda function as predicate)